src/convertToObject.js

```diff
@@ -12,7 +13,21 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const obj = {};
+  const arr = sourceString
+    .split(';')
+    .map(el => el.trim())
+    .filter(el => el.length !== 0);
+
+  arr.forEach(element => {
+    const index = element.split('').findIndex(el => el === ':');
+    const key = element.slice(0, index).trim();
+    const value = element.slice(index + 1).trim();
```

Comment:
in my opinion, 'index' is redundant here. you can avoid this using only split(':')[index] for key and value